visualization/app/codeCharta/util/lazyLoader.ts

```diff
@@ -0,0 +1,50 @@
+import { isStandalone } from "./envDetector"
+
+export class LazyLoader {
+	private static _fileName: string
+	private static _nodePath: string
+
+	static async openFile(fileName: string = LazyLoader._fileName, nodePath: string = LazyLoader._nodePath) {
+		if (!isStandalone()) {
+			return
+		}
+		LazyLoader._fileName = fileName
+		LazyLoader._nodePath = nodePath
+		const directory = localStorage.getItem(LazyLoader._fileName)
+		//replace '/root' with directory
+		const path = nodePath.replace(nodePath.split("/", 2).join("/"), directory)
+		if (directory === null) {
+			return LazyLoader.setDirectory()
+		}
+		//check if root still exists
+		const exists = await LazyLoader.checkDirExists(directory)
+		if (!exists) {
+			alert(`Unknown or non-existent directory: ${directory}`)
+			return LazyLoader.setDirectory()
+		}
+		const open = await import("open")
+		try {
+			await open.default(`file:///${path}`)
+		} catch (error) {
+			alert(error)
```

Comment:
I personally would rather use our internal error handling. @NearW do you know how to use that here?

It's not a blocker though.

Reply:
I think the idea is to throw an error dialog to the user to provide feedback that it couldn't be opened?

Reply:
> I think the idea is to throw an error dialog to the user to provide feedback that it couldn't be opened?

Right. However, as discussed before, there is a problem with `async await` while using two dialogs (error and prompt) at the same time.